src/hooks/useOmitBy.ts

```diff
@@ -0,0 +1,15 @@
+import { useCallback } from "react";
+
+export default function useOmitBy<T extends Record<string, unknown>>(
+  predicate: (value: unknown) => boolean = (value) =>
+    value !== "" && value !== undefined && value !== null,
+) {
+  return useCallback(
+    (obj: T): Partial<T> => {
+      return Object.fromEntries(
+        Object.entries(obj).filter(([_, value]) => predicate(value)),
+      ) as Partial<T>;
+    },
+    [predicate],
+  );
+}
```

Comment:
i don't see a reason why this needs to be a hook. can't this just be simply a utility function? code would be much cleaner that way.